healthScore/tests.py

```diff
@@ -354,100 +348,151 @@ def test_view_history_requests(self):
 
-    def test_registration_not_post_method(self):
-        url = reverse("registration")
-        request_struct = {
-            "email": "test@test.com",
-            "username": "testUser",
-            "fullname": "test user",
-            "dob": "1990-02-02",
-            "gender": "male",
-            "street_address": "123 foo bar, foo",
-            "city": "NY",
-            "state": "NY",
-            "phone_number": "1235467890",
-        }
-        request = self.factory.put(url, request_struct)
-        response = registration(request)
-        self.assertEqual(response.status_code, 404)
+class HomepageViewTest(TestCase):
+    def test_homepage_view(self):
+        response = self.client.get(reverse("homepage"))  # test the view
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "homepage.html")
 
-    def test_registration_user_email_exists_error(self):
-        url = reverse("registration")
-        request_struct = {
-            "email": "user1@example.com",
-            "username": "testUser",
-            "fullname": "test user",
-            "dob": "1990-02-02",
-            "gender": "male",
-            "street_address": "123 foo bar, foo",
-            "city": "NY",
-            "state": "NY",
-            "phone_number": "1235467890",
-        }
-        request = self.factory.post(url, request_struct)
-        response = registration(request)
-        self.assertEqual(response.status_code, 500)
 
-    def test_registration_user_username_exists_error(self):
-        url = reverse("registration")
-        request_struct = {
-            "email": "test2@test2.com",
-            "username": "user1",
-            "fullname": "test user",
-            "dob": "1990-02-02",
-            "gender": "male",
-            "street_address": "123 foo bar, foo",
-            "city": "NY",
-            "state": "NY",
-            "phone_number": "1235467890",
-        }
-        request = self.factory.post(url, request_struct)
-        response = registration(request)
-        self.assertEqual(response.status_code, 500)
+class LoginViewTest(TestCase):
+    def setUp(self):
+        self.user = User.objects.create_user(
+            email="test@example.com", password="testpassword"
+        )
+
+    def test_login_view(self):
+        response = self.client.get(reverse("login"))
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "login.html")
+
+    def test_post_request_valid_credentials(self):
+        response = self.client.post(
+            reverse("login"), {"email": "test@example.com", "password": "testpassword"}
+        )
+        self.assertRedirects(response, reverse("homepage"))
+
+    def test_post_request_invalid_credentials(self):
+        response = self.client.post(
+            reverse("login"), {"email": "test@example.com", "password": "wrongpassword"}
+        )
+        self.assertEqual(response.status_code, 200)
```

Comment:
Won't the invalid credentials result in a status code which is not 200? 200 would be for a login "success" instead. Could you please check the status code once?

Reply:
200 means login page has been rendered successfully. No matter the user logs in successfully or not, the page should always be 200. For error message, it will show on that log in page to remind the users

Reply:
Ohhh